ruby/decorators.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0..10]
```

Comment:
- [ ] kindly consider the name has a maximum of 10 characters. If it's longer it should trim the word.
> It will be something like this 
```ruby
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

Reply:
Thanks very much @omarsalem7, I will implement this immediately,

please could I know why your method works better? as I am still trying to grasp ruby concepts.